Sdtrig.tex

```diff
@@ -26,16 +27,18 @@ \chapter{Sdtrig (ISA Extension)}
 
 \section{Enumeration}
 
-\begin{steps}{Each trigger may support a variety of features. A debugger can
+\begin{steps}{Each trigger may support a variety of features.
+    If Sdtrig is implemented or unimplemented CSRs raise an exception,
```

Comment:
Is it possible for Sdtrig to be implemented but with zero triggers?  That seems kind of silly but I think that technically somebody could claim that their zero trigger core is Sdtrig complliant.  That serves no real purpose but looks slightly more impressive in marketing Powerpoint (which is slimy but technically legit).  To avoid this, I think that we should have a sentence somewhere that says that an Sdtrig-compliant implementation has at least one trigger (unless I'm forgetting some Sdtrig feature other than triggers that somebody might want to implement without doing triggers).  That prevents any Powerpoint ethical gray areas and also makes this sentence correct.

Reply:
That was explicitly allowed back when there was a single debug extension, to keep triggers optional. You're right that it doesn't make sense to allow Sdtrig to be implemented with 0 triggers. I support the idea, but I don't think that fits in this PR, and it's also technically backwards incompatible.

Reply:
I don't think that it's incompatible.  Old implementations would have said that they do the debug spec but there was no "Sdtrig" until recently.  I guess that somebody in the last year or so could have claimed Sdtrig compliance without any triggers but that seems unlikely and I don't think that adding this new requirement would require them to change anything other than some Powerpoint slides.

This is along the lines of what I was talking about last week when I suggested explicitly saying that you can do external debug without Sdtrig, but you pointed out the diagram with the dotted line around the TM.

If we don't say that Sdtrig compliance => non-zero number of triggers then I worry that this recipe doesn't work for implementations that don't get exceptions on accesses to unimplemented CSRs.  (In real life, I hope that everyone gets exceptions AND has at least one trigger if they claim compliance, but in theory there's a problem.)

Reply:
I made #872 for this.